### PR TITLE
Ensure that the channel/connection is for the specified source and destination chains when performing the query

### PR DESCRIPTION
**Pull request by brapse**
Closes: #873

## Description

Ensure that the channel/connection is for the specified source and destination chains when performing the query.

______

For contributor use:

- [x] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ Jongwhan Lee (@leejw51crypto) ([#878]).
 
 - [ibc-relayer-cli]
   - Fix for `ft-transfer` mismatching arguments ([#869])
+  - Fix channel destination chain mismatch on unreceived-packets or unreceived-acks ([#873])
```
Comment (romac) [inline CHANGELOG.md:39]: Can you also please add a corresponding link in the section below (line 59)?

**Review by romac — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (brapse) [reply CHANGELOG.md:39]: yus 👍🏼

**Comment by romac** [comment]
🥇

File: relayer-cli/src/commands/query/packet.rs [hunk 2]
```diff
@@ -254,16 +260,54 @@ impl Runnable for QueryUnreceivedPacketsCmd {
...
+            Some(id) => id,
+            None => {
+                return Output::error(format!(
+                    "no connection hops for channel '{}'",
+                    self.src_channel_id
+                ))
+                .exit()
+            }
+        };
+
```
Comment (ancazamfir) [inline relayer-cli/src/commands/query/packet.rs:280]: I think we can remove this and query channel above.

File: relayer-cli/src/commands/query/packet.rs [hunk 3]
```diff
@@ -548,16 +591,55 @@ impl Runnable for QueryUnreceivedAcknowledgementCmd {
...
+        let src_connection_id = match channel.connection_hops().first() {
+            Some(id) => id,
+            None => {
+                return Output::error(format!(
+                    "no connection hops for channel '{}'",
+                    self.src_channel_id
+                ))
+                .exit()
+            }
+        };
```
Comment (ancazamfir) [inline relayer-cli/src/commands/query/packet.rs:611]: same here.

File: relayer-cli/src/commands/query/packet.rs (hunk 2 above)
Comment (brapse) [reply relayer-cli/src/commands/query/packet.rs:280]: What are the advantages of querying the channel vs pulling the `src_connection_id` from the `channel` data structure already available in memory?

**Pull request merged**